src/transformers/tokenization_utils_base.py

```diff
@@ -1818,10 +1818,22 @@ def convert_added_tokens(obj: Union[AddedToken, Any]):
             added_tok_encoder_sorted = list(sorted(added_tok_encoder.items(), key=lambda x: x[1]))
 
             for token, index in added_tok_encoder_sorted:
-                assert index == len(tokenizer), (
-                    f"Non-consecutive added token '{token}' found. "
-                    f"Should have index {len(tokenizer)} but has index {index} in saved vocabulary."
-                )
+                if has_tokenizer_file and index != len(tokenizer) and tokenizer.convert_tokens_to_ids(token) != index:
+                    # Tokenizer fast: added token needs to either be in the vocabulary with the proper index or the
+                    # index is the current length of the tokenizer (not in vocabulary)
+                    raise ValueError(
+                        f"Wrong index found for {token}: should be {tokenizer.convert_tokens_to_ids(token)} but found "
+                        f"{index}."
+                    )
```

Comment:
This part is added for fast tokenizers: if the index of the token is not `len(tokenizer)` it's probably because the added tokens was already in the tokenizer.json file and in thi case we just check it has the right index.